nekoyume/Assets/_Scripts/UI/Widget/Collection.cs

```diff
@@ -368,5 +447,156 @@ private static bool IsInToggle(StatModifier stat, StatType toggleStatType)
+                case ESortType.Level:
+                    return L10nManager.Localize("UI_LEVEL");
+            }
+
+            return string.Empty;
+        }
+
+        private void OnSortDropdownValueChanged(int index)
+        {
+            _sortType = (ESortType)index;
+            UpdateItems();
+        }
+
+        private int SortCollection(CollectionModel a, CollectionModel b)
+        {
+            if (a == null && b == null)
+                return 0;
+            if (a == null) return 1;
+            if (b == null) return -1;
+
+            // 1. 활성화 가능
+            if (a.CanActivate != b.CanActivate)
+                return a.CanActivate ? -1 : 1;
+
+            // 2. 재료 일정 부분 달성
+            var aPartiallyActive = a.Materials.Any(CheckHasMaterialWhenInactive);
+            var bPartiallyActive = b.Materials.Any(CheckHasMaterialWhenInactive);
+            if (aPartiallyActive != bPartiallyActive)
+                return aPartiallyActive ? -1 : 1;
+
+            // 3. 재료 모두 미달성 (나머지)
+
+            // 설정된 타입별로 정렬
+            var sortByTypeValue = SortByType(a, b, _sortType);
+            if (sortByTypeValue != 0)
+                return sortByTypeValue;
+
+            // 다른 조건이 같다면 ID로 비교
+            if (a.Row.Id != b.Row.Id)
+                return a.Row.Id < b.Row.Id ? -1 : 1;
+
+            return 0;
+        }
+
+        private int SortByType(CollectionModel a, CollectionModel b, ESortType type)
+        {
+            var sortTypeWeight = _isSortDescending ? 1 : -1;
+            switch (_sortType)
+            {
+                case ESortType.Grade:
+                    var aGrade = a.Materials.Max(material => material.Grade);
+                    var bGrade = b.Materials.Max(material => material.Grade);
+                    return (bGrade - aGrade) * sortTypeWeight;
+                case ESortType.Level:
+                    var aLevel = a.Materials.Max(material => material.EnoughLevel);
+                    var bLevel = b.Materials.Max(material => material.EnoughLevel);
+                    return (bLevel - aLevel) * sortTypeWeight;
+            }
+            return 0;
+        }
+
+        private bool CheckHasMaterialWhenInactive(CollectionMaterial material)
+        {
+            if (material == null)
+                return false;
+
+            return material.HasItem && !material.Active;
+        }
+        #endregion Sort
+
+        #region Filter
+        private readonly List<CollectionModel> _filteredItems = new();
+
+        private bool IsNeedSearch => !string.IsNullOrWhiteSpace(_searchInputField.text);
+
+        private List<CollectionModel> RefreshFilteredItems()
+        {
+            _filteredItems.Clear();
+
+            foreach (var model in _items)
+            {
+                bool isContained = !(IsNeedSearch && !IsMatchedSearch(model));
+
+                if (isContained)
+                    _filteredItems.Add(model);
+            }
```

Comment:
```suggestion
            _filteredItems.AddRange(_items.Where(model => !IsNeedSearch || IsMatchedSearch(model)));
```

이건 어떤가요? (가독성이 떨어진다 싶으면 스루하셔도 됩니다.)